lib/src/views/Calendar/CalendarView.tsx

```diff
@@ -0,0 +1,198 @@
+import * as React from 'react';
+import CalendarHeader from './CalendarHeader';
+import { Calendar } from '../..';
+import { CalendarProps } from './Calendar';
+import { YearSelection } from './YearSelection';
+import { makeStyles } from '@material-ui/styles';
+import { MonthSelection } from './MonthSelection';
+import { DatePickerView } from '../../DatePicker';
+import { SlideDirection } from './SlideTransition';
+import { useUtils } from '../../_shared/hooks/useUtils';
+import { ParsableDate } from '../../constants/prop-types';
+import { MaterialUiPickersDate } from '../../typings/date';
+import { CircularProgress, Grid } from '@material-ui/core';
+import { FadeTransitionGroup } from './FadeTransitionGroup';
+import { useParsedDate } from '../../_shared/hooks/useParsedDate';
+
+export interface CalendarViewProps extends Omit<CalendarProps, 'minDate' | 'maxDate'> {
+  date: MaterialUiPickersDate;
+  view: DatePickerView;
+  changeView: (nextView: DatePickerView) => void;
+  onChange: (date: MaterialUiPickersDate, isFinish?: boolean) => void;
+  /** Min date @DateIOType */
+  minDate?: ParsableDate;
+  /** Max date */
+  maxDate?: ParsableDate;
+}
+
+export type ExportedCalendarProps = Omit<
+  CalendarViewProps,
+  'date' | 'view' | 'onChange' | 'changeView' | 'slideDirection' | 'currentMonth'
+>;
+
+export type ReducerAction<TType, TAdditional = {}> = { type: TType } & TAdditional;
+
+interface ChangeMonthPayload {
+  direction: SlideDirection;
+  newMonth: MaterialUiPickersDate;
+}
+
+function calendarStateReducer(
+  state: {
+    loadingQueue: number;
+    currentMonth: MaterialUiPickersDate;
+    slideDirection: SlideDirection;
+  },
+  action:
+    | ReducerAction<'popLoadingQueue'>
+    | ReducerAction<'changeMonth', ChangeMonthPayload>
+    | ReducerAction<'changeMonthLoading', ChangeMonthPayload>
+) {
+  switch (action.type) {
+    case 'changeMonthLoading': {
+      return {
+        ...state,
+        loadingQueue: state.loadingQueue + 1,
+        slideDirection: action.direction,
+        currentMonth: action.newMonth,
+      };
+    }
+    case 'changeMonth': {
+      return {
+        ...state,
+        slideDirection: action.direction,
+        currentMonth: action.newMonth,
+      };
+    }
+    case 'popLoadingQueue': {
+      return {
+        ...state,
+        loadingQueue: state.loadingQueue <= 0 ? 0 : state.loadingQueue - 1,
+      };
+    }
+  }
+}
+
+const useClasses = makeStyles({
+  viewTransitionContainer: {
+    overflowY: 'auto',
+  },
+  gridFullHeight: {
+    flex: 1,
+    height: '100%',
+  },
+});
+
+export const CalendarView: React.FC<CalendarViewProps> = ({
+  date,
+  view,
+  onChange,
+  changeView,
+  onMonthChange,
+  minDate: unparsedMinDate,
+  maxDate: unparsedMaxDate,
+  loadingIndicator = <CircularProgress />,
+  ...rest
```

Comment:
For consistency with the core?

```suggestion
  ...other
```